### PR TITLE
Updated Error and Exception handling

### PR DESCRIPTION
**Pull request by deoyani**
Added Exception Handler along with various Exception classes.
Added a small validator for input data ids.
Updated exception handling in Download methods

### CODE REVIEW

**Comment by deoyani** [comment]
@RayPlante  Please check as per your convenience. Let me know if there are any issues.

**Comment by deoyani** [comment]
@RayPlante I added extra check for RMM if record not  found in downloadAll function.

File: src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	@ResponseStatus(HttpStatus.CONFLICT)
+ 	  @ExceptionHandler(JsonMappingException.class)
+	  @ResponseBody
+	  /**
+	   * Handles General Exception
+	   * @param exception
+	   * @return ErrorInfo object with error details
+	   */
+	  public ErrorInfo  myErrors(JsonMappingException exception) {
+	    logger.info("----Caught IOException----\n"+request.getDescription(false)+"\n Detail IOException:"+exception.getMessage()+exception.getStackTrace());
```
Comment (RayPlante) [inline src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java]: "IOException":  Cut and paste error?

File: src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	  @ResponseStatus(HttpStatus.CONFLICT)
+	  @ExceptionHandler(Exception.class)
+	  @ResponseBody
+	  /**
+	   * Handles General Exception
+	   * @param exception
+	   * @return ErrorInfo object with error details
+	   */
+	  public ErrorInfo  myError(Exception exception) {
+	    logger.info("----Caught Exception----\n"+request.getDescription(false)+"\n Detail IOException:"+exception.getMessage()+exception.getStackTrace());
```
Comment (RayPlante) [inline src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java]: IOException:  Cut-and-paste error?

File: src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	   * @param ResourceNotFoundException
+	   * @return ErrorInfo object with error details
+	   */
+	  public ErrorInfo resourceNotFound(ResourceNotFoundException exception) {
+            logger.info("----Caught KeywordNotFoundException----\n"+request.getDescription(false)+"\n Detail ResourceNotFoundException:"+exception.getMessage()+exception.getStackTrace());
+            return new ErrorInfo(request.getContextPath(), "Requested record/data not found.",HttpStatus.NOT_FOUND.toString());
+            
+	  }
+	  
+	  @ResponseStatus(HttpStatus.BAD_REQUEST)
```
Comment (RayPlante) [inline src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java]: BAD_REQUEST (400) is for an error that is the client's fault--they called the service incorrectly.  An IOException sounds like a server problem (e.g. the storage system had a read error), yes?  This should result in `HttpStatus.INTERNAL_SERVER_ERROR`, yes?

Comment (deoyani) [reply src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java]: yes.

Comment (deoyani) [reply src/main/java/gov/nist/oar/ds/controller/GlobalExceptionHandler.java]: Its just by mistake.

**Comment by deoyani** [comment]
Updated messages properly according to the error messages.

**Comment by RayPlante** [comment]
@deoyani This branch is currently deployed on oardev.  When I access this URL:
```
curl -vk https://oardev.nist.gov/od/ds//ECBCC1C1300D2ED9E04306570681B10719/goober.txt
```
i.e. legal/existing dataset id but non-existant file, I get a 500 status back and a JSON body that indicates a `FileNotFoundException` was raised.   The log shows the info message from the general exception handler for FileNotFoundException; however, it also shows that another exception gets raised that may preventing the right status to be returned:
```
HttpMediaTypeNotAcceptableException: Could not find acceptable representation
```

**Comment by deoyani** [comment]
@RayPlante Looking into it.

**Pull request merged**